common/beerocks/bwl/dummy/base_wlan_hal_dummy.cpp

```diff
@@ -280,6 +280,24 @@ bool base_wlan_hal_dummy::set(const std::string &param, const std::string &value
 
 bool base_wlan_hal_dummy::ping() { return true; }
 
+bool base_wlan_hal_dummy::write_status_file(const std::string &filename, const std::string &value)
+{
+    auto full_path{get_status_dir() + "/" + filename};
+    auto full_path_tmp{full_path + ".tmp"};
+    std::ofstream statusfile(full_path_tmp);
+    statusfile << value;
+    if (!statusfile) {
+        LOG(ERROR) << "Failed writing to " << full_path;
+        return false;
+    }
+    statusfile.close();
```

Comment:
We have to close the file before the rename. The whole point of using a temporary file is to make sure it's atomic, i.e. the file is either the old one or the new one, not something in-between. If I don't close the file, then it's possible that the rename happens before the data is actually flushed to the file, and that a reader in another process sees an empty file.

Note that to be fully atomic, we should actually also do an fsync on the directory before the rename, but that's just making sure that the metadata is updated and we're not that interested in the metadata.

Also, we should be doing an fsync on the file itself. But again in this particular case it's not needed because we don't need atomicity over reboots, only between processes.

It might actually be sufficient to do a flush rather than close, but I feel safer this way.

And finally: the same could also be achieved by putting the statusfile in an inner scope, so its lifetime ends at this point. However, I think that that is even more confusing and error prone than an explicit clone.

That said, a comment would really be a good idea here.